src/tinuous/__main__.py

```diff
@@ -76,6 +77,29 @@ def from_travis_event(cls, travis_event: str) -> Optional["EventType"]:
         }.get(travis_event)
 
 
+class APIClient:
+    def __init__(self, base_url: str, headers: Dict[str, str]):
+        self.base_url = base_url
+        self.session = requests.Session()
+        self.session.headers.update(headers)
+
+    def get(self, path: str, **kwargs: Any) -> requests.Response:
+        if path.lower().startswith(("http://", "https://")):
+            url = path
+        else:
+            url = self.base_url.rstrip("/") + "/" + path.lstrip("/")
+        while True:
```

Comment:
let's not try forever - wouldn't be good for a cron script to just halting there upon hitting some "proper" server side issue.  Do e.g. up to 10 tries but make sleep e.g. `sleep(trial)` so it is not bombarding each second.  If no luck -- crash upon the last one.

Reply:
Updated.